### DISCUSSION TITLE
Can I service static html with include_str! macro (ie include it in my binary)?

### DISCUSSION THREAD

**Discussion by AnatolyBuga** [actix/actix-web#2989] [Q&A] [answered]
Hello and thank you for this amazing lib!

At the moment I include a static index.html in my App with **last** service call in the App builder like this:
`.service(fs::Files::new("/", &static_files_dir).index_file("index.html"))`.

`factory` here is an instance of `actix_files::Files`, and not a str. 

_**I would like to include the file (`index.html`) in my binary**_, so that I can distribute this to the client, and they wouldn't have this `index.html` file on their machines. For that I suppose I could use `include_str!` macro, but I'd need to service this str somehow. 

Can I achieve this?

**Comment by robjtede** [comment, marked as answer]
```rust
let index = include_str!("./index.html");

HttpResponse::Ok()
  .content_type(ContentType::html())
  .body(index)

# or from actix-web-lab
Html::new(index)
```

If there's more files you want to include statically, there's this lib: https://docs.rs/actix-web-static-files
